ironfish/src/rpc/clients/socketClient.ts

```diff
@@ -2,60 +2,53 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 import net from 'net'
-import { IpcClient } from 'node-ipc'
 import { Assert } from '../../assert'
 import { Event } from '../../event'
-import { Logger } from '../../logger'
-import { PromiseUtils, SetTimeoutToken, YupUtils } from '../../utils'
+import { createRootLogger, Logger } from '../../logger'
+import { ErrorUtils, PromiseUtils, SetTimeoutToken, YupUtils } from '../../utils'
 import {
+  MESSAGE_DELIMITER,
+  ServerSocketRpc,
+  ServerSocketRpcSchema,
   SocketRpcErrorSchema,
   SocketRpcResponseSchema,
   SocketRpcStreamSchema,
 } from '../adapters'
+import { MessageBuffer } from '../messageBuffer'
 import { isRpcResponseError, RpcResponse } from '../response'
 import { Stream } from '../stream'
 import { RpcClient } from './client'
 import {
   RequestTimeoutError,
   RpcConnectionError,
   RpcConnectionLostError,
+  RpcConnectionRefusedError,
   RpcRequestError,
 } from './errors'
 
-const REQUEST_TIMEOUT_MS = null
-
-export type RpcClientConnectionInfo =
-  | {
-      mode: 'ipc'
-      socketPath: string
-    }
-  | {
-      mode: 'tcp'
-      host: string
-      port: number
-    }
+export type RpcSocketClientConnectionInfo = {
+  path?: string
+  host?: string
+  port?: number
+}
```

Comment:
I think this is more confusing than the previous RpcClientConnectionInfo type -- on the naming, "path" could be a combination of "host" and "port". And I think the correct usage of RpcClientConnectionInfo is to specify either fields for ipc-mode, or fields for tcp-mode, and the previous type made it clear which fields were relevant to which connection mode.